### PR TITLE
[plugin.video.dailymotion_com] v2.4.1

### PR DESCRIPTION
**Pull request by Gujal00**
### Checklist:
- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.dailymotion_com/resources/lib/dailymotion.py [hunk 1]
```diff
@@ -0,0 +1,798 @@
...
+    addDir(translation(30036), urlMain + "/user/" + dmUser + "/subscriptions?fields=description,duration,id,owner.username,taken_time,thumbnail_large_url,title,views_total&sort=recent&limit=" + itemsPerPage + "&family_filter=" + familyFilter + "&localization=" + language + "&page=1", 'listVideos', _ipath + "following.png")
+    addDir(translation(30037), urlMain + "/user/" + dmUser + "/favorites?fields=description,duration,id,owner.username,taken_time,thumbnail_large_url,title,views_total&sort=recent&limit=" + itemsPerPage + "&family_filter=" + familyFilter + "&localization=" + language + "&page=1", 'listVideos', _ipath + "favourites.png")
+    addDir(translation(30038), urlMain + "/user/" + dmUser + "/playlists?fields=id,name,videos_total&sort=recent&limit=" + itemsPerPage + "&family_filter=" + familyFilter + "&localization=" + language + "&page=1", 'listUserPlaylists', _ipath + "playlists.png")
+    xbmcplugin.setContent(pluginhandle, 'addons')
+    if force_mode:
```
Comment (enen92) [inline plugin.video.dailymotion_com/resources/lib/dailymotion.py]: not allowed sorry. Don't know how this was missed before :/

File: plugin.video.dailymotion_com/resources/lib/dailymotion.py [hunk 2]
```diff
@@ -0,0 +1,798 @@
...
+
+
+def sortVideos2(url):
+    addDir(translation(30010), url.replace("sort=visited", "sort=visited-hour"), "listVideos", _ipath + "most_viewed.png")
+    addDir(translation(30011), url.replace("sort=visited", "sort=visited-today"), "listVideos", _ipath + "most_viewed.png")
+    addDir(translation(30012), url.replace("sort=visited", "sort=visited-week"), "listVideos", _ipath + "most_viewed.png")
+    addDir(translation(30013), url.replace("sort=visited", "sort=visited-month"), "listVideos", _ipath + "most_viewed.png")
+    addDir(translation(30014), url, 'listVideos', _ipath + "most_viewed.png")
+    xbmcplugin.setContent(pluginhandle, 'addons')
+    if force_mode:
```
Comment (enen92) [inline plugin.video.dailymotion_com/resources/lib/dailymotion.py]: same

File: plugin.video.dailymotion_com/resources/lib/dailymotion.py [hunk 3]
```diff
@@ -0,0 +1,798 @@
...
+            count += 1
+        else:
+            addLink(title, vid, 'playVideo', thumb.replace("\\", ""), user, desc, duration, date, count)
+            count += 1
+    if content['has_more']:
+        currentPage = content['page']
+        nextPage = currentPage + 1
+        addDir(translation(30001) + " (" + str(nextPage) + ")", url.replace("page=" + str(currentPage), "page=" + str(nextPage)), 'listVideos', _ipath + "next_page2.png")
+    xbmcplugin.setContent(pluginhandle, "episodes")
+    if force_mode:
```
Comment (enen92) [inline plugin.video.dailymotion_com/resources/lib/dailymotion.py]: same

File: plugin.video.dailymotion_com/resources/lib/dailymotion.py [hunk 4]
```diff
@@ -0,0 +1,798 @@
...
+            thumb = item['avatar_large_url']
+            videos = item['videos_total']
+            views = item['views_total']
+            addUserDir(user, 'owner:' + user, 'sortVideos1', thumb.replace("\\", ""), "Views: " + str(views) + "\nVideos: " + str(videos))
+    if content['has_more']:
+        currentPage = content['page']
+        nextPage = currentPage + 1
+        addDir(translation(30001) + " (" + str(nextPage) + ")", url.replace("page=" + str(currentPage), "page=" + str(nextPage)), 'listUsers', _ipath + "next_page.png")
+    xbmcplugin.setContent(pluginhandle, "addons")
+    if force_mode:
```
Comment (enen92) [inline plugin.video.dailymotion_com/resources/lib/dailymotion.py]: same

File: plugin.video.dailymotion_com/resources/lib/dailymotion.py [hunk 5]
```diff
@@ -0,0 +1,798 @@
...
+        vid = item['id']
+        thumb = item['thumbnail_large_url']
+        views = item['views_last_hour']
+        addLiveLink(title, vid, 'playLiveVideo', thumb.replace("\\", ""), views)
+    if content['has_more']:
+        currentPage = content['page']
+        nextPage = currentPage + 1
+        addDir(translation(30001) + " (" + str(nextPage) + ")", url.replace("page=" + str(currentPage), "page=" + str(nextPage)), 'listLive', _ipath + "next_page2.png")
+    xbmcplugin.setContent(pluginhandle, "episodes")
+    if force_mode:
```
Comment (enen92) [inline plugin.video.dailymotion_com/resources/lib/dailymotion.py]: same

File: plugin.video.dailymotion_com/resources/lib/dailymotion.py (hunk 1 above)
Comment (Gujal00) [reply plugin.video.dailymotion_com/resources/lib/dailymotion.py]: @enen92 
Yes this was raised before and clarified by @Rechi that by default modes can not be set and only set if enabled in settings by user. The code follows that principle and the default is not setting any modes as per that principle
Discussion from #2400 
``` Rechi on May 27 Member
It's okay to allow add a configure setting for forcing the view mode, but it has to be off by default.
```

**Comment by Gujal00** [comment]
> 
> 
> I'm sorry but setting view modes is not allowed in Kodi plugins. See point 5: https://kodi.wiki/view/Add-on_rules

@enen92 
Yes this was raised before and clarified by @Rechi that by default modes can not be set and only set if enabled in settings by user. The code follows that principle and the default is not setting any modes as per that principle
Discussion from #2400 
``` Rechi on May 27 Member
It's okay to allow add a configure setting for forcing the view mode, but it has to be off by default.
```

File: plugin.video.dailymotion_com/resources/lib/dailymotion.py (hunk 1 above)
Comment (enen92) [reply plugin.video.dailymotion_com/resources/lib/dailymotion.py]: We clarified and it's okay to keep it default off

Comment (Gujal00) [reply plugin.video.dailymotion_com/resources/lib/dailymotion.py]: Thanks

**Pull request merged**